<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-15">
<title>GeneSrF</title>

<script type="text/javascript" language="javascript">
	function getArgs(  ) {
	    var args = new Object(  );
	    var query = location.search.substring(1);      // Get query string
	    var pairs = query.split("&");                  // Break at &
	    for(var i = 0; i < pairs.length; i++) {
	        var pos = pairs[i].indexOf('=');          // Look for "name=value"
	        if (pos == -1) continue;                  // If not found, skip
	        var argname = pairs[i].substring(0,pos);  // Extract the name
	        var value = pairs[i].substring(pos+1);    // Extract the value
	        args[argname] = unescape(value);          // Store as a property
       						  // In JavaScript 1.5, use decodeURIComponent(  ) 
					          // instead of escape(  )
    	    }
	    return args;     // Return the object
	}	

</script>
</head>

<body>
<BODY background="images/fondo3.jpg">

<form action="http://genesrf2.bioinfo.cnio.es/cgi-bin/genesrfR.cgi"
enctype="multipart/form-data" method="POST">




<center>
<table border=0 cellspacing=0 width="100%">
  <tr bgcolor="#54C571">
    <td rowspan="3" align=center width=140>
      <a href="http://genesrf2.bioinfo.cnio.es">
      <img SRC="genesrf_logo.gif"
      border=0
      alt="genesrf">
      </a>
    </td>
    <td bgcolor="#254117" align=center width="100%">
      <a href="http://bioinfo.cnio.es" STYLE="text-decoration:none">
      <font color=white>
      <b>Bioinformatics Unit</b>
      </font>
      </a>
        -  
      <a href="http://www.cnio.es" STYLE="text-decoration:none">
      <font color=white>
      <b>CNIO</b>
      </font>
      </a>
    </td>
    <td rowspan="3" align=center width=124>
      <a href="http://bioinfo.cnio.es">
      <img src="/images/bioinfo.png" border=0>
      </a>
    </td>
  </tr>
  <tr>
    <td bgcolor="#54C571" align=center valign=middle  STYLE="font-size:12">
      <br>
      <font size=5>
      <b>GeneSrF</b>
      </font>
      <br> 
    </td>
  </tr>
  <tr>
    <td bgcolor="#254117" align=center>
       
    </td>
  </tr>
</table>
</center>
<br>


<h2>GeneSrF: gene selection with random forests (v. 20070524)</h2>

<p style="font-family: 'Bitstream Vera Sans',Verdana,sans; font-size: 9pt"> GeneSrF is a
    web tool for gene selection in classification problems that uses random
    forest. Two approaches for gene
    selection are used: one is targeted towards identifying small,
    non-redundant sets of genes that have good predictive performance. The
    second is a more heuristic graphical approach that can be used to identify
    large sets of genes (including redundant genes) related to the outcome of
    interest. The first approach is described in detail in this
<a
href="http://www.biomedcentral.com/1471-2105/7/3">paper</a>. The R code is available as an R package from
<a href="http://cran.r-project.org/src/contrib/Descriptions/varSelRF.html">CRAN</a> or
from this <a
href="http://www.ligarto.org/rdiaz/Software/Software.html#varSelRF">link</a>.
For further details see the
<a href="./help/genesrf-help.html" target="help_window">help</a>.</p>



<p style="font-family: 'Bitstream Vera Sans',Verdana,sans; font-size: 9pt"">To use this web
tool you need to provide two files, one with the gene expression data and one with
the class labels, and press "Submit".</p>
<hr>

<h3>Input files</h3>

<TABLE>
<tr><td>
<b>Expression data file:</b> (<a href="./help/genesrf-help.html#input"
target="help_window">?</a>) 
</td><td>
<script Language="Javascript">
  var args = getArgs();
  if (args.datadir) {
    document.write("<span style=\"color: red\">Input Data from <i>pre</i>P</b></span><input type=\"hidden\" value =" + args.datadir + "  name=\"covariate2\">");
   }else {
  	document.write("<input type=\"FILE\" name=\"covariate\" size = \"40\" maxlength =\"200\">");
   }
</script>
</td></tr>
<!--     <p style="font-family: 'Bitstream Vera Sans',Verdana,sans"> -->
<!-- A tab-separated text file with the covariates; generally the gene  -->
<!--     expression data. In this file, rows represent variables (generally genes), -->
<!--     and columns represent subjects, or samples, or arrays. If you want to -->
<!--     <b>name your arrays</b> place a line that starts with "&#035;", after -->
<!--     the "&#035;" put "Name" or "NAME" or "name" and write the array names -->
<!--     (separated by tabs).  The <b>first column</b> is assumed to contain the ID -->
<!--     information for genes, marker, or whatever. This will be used to label the -->
<!--     output and thus this first column is not used in the -->
<!--     analyses. Note that <b>gene names and array names MUST be unique</b> and -->
<!--       <b>missing values are NOT allowed</b>. For further details see the -->
<!-- <a href="./help/genesrf-help.html#input" target="help_window">help</a>.</p> -->

<tr><td>
<b>Class file: </b> (<a href="./help/genesrf-help.html#input" target="help_window">?</a>)
</td><td>
<input type="FILE" name="class", size = "40" maxlength = "200">
</td></tr>
</table>
<!-- <p style="font-family: 'Bitstream Vera Sans',Verdana,sans"> -->
<!-- A tab-separated text file with the class labels (e.g., healthy or affected, or -->
<!-- different types of cancer) that group the samples. The random forests built -->
<!-- will try to predict these class labels.  For further details see the -->
<!-- <a href="./help/genesrf-help.html#input" target="help_window">help</a>.</p> -->



<hr/ >
<h3>Type of gene identifier and species</h3>
<p style="font-family: 'Bitstream Vera Sans',Verdana,sans; font-size: 9pt">
This information is used to provide clickable links (that take you to the one-query-version of our 
<a href="http://idconverter.bioinfo.cnio.es">IDConverter</a>) in the results.</p>

<table>
  <tr><td width="100"></td><td width="400"><b>Type of ID for gene names</b></td>
    <td width="400"><b>Organism</b></td></tr>
  <tr><td></td><td>
      <input type="RADIO" name="idtype" value="affy">  Affymetrix ID</input>
    </td><td>
      <input type="RADIO" name="organism" value="Hs">  Human (<i>Homo sapiens</i>)</input>
  </td></tr>
  
  <tr><td></td><td>
      <input type="RADIO" name="idtype" value="clone">  Clone ID (IMAGE Consortium)</input>
    </td><td>
      <input type="RADIO" name="organism" value="Mm">  Mouse (<i>Mus musculus</i>)</input>
  </td></tr>
  <tr><td></td><td>
      <input type="RADIO" name="idtype" value="acc">  GenBank Accession</input>
    </td><td>
      <input type="RADIO" name="organism" value="Rn">  Rat (<i>Rattus norvegicus</i>)</input>
  </td></tr>
  <tr><td></td><td>
      <input type="RADIO" name="idtype" value="ensembl">  Ensembl Gene</input>
    </td><td>
      <input type="RADIO" name="organism" checked value="None">  None of the above</input>
  </td></tr>
  <tr><td></td><td>
      <input type="RADIO" name="idtype" value="ug">  Unigene cluster</input>
    </td><td></td></tr>
  <tr><td></td><td>
      <input type="RADIO" name="idtype" value="rsrna">  RefSeq RNA</input>
    </td><td></td></tr>
  <tr><td></td><td>
      <input type="RADIO" name="idtype" value="rspeptide">  RefSeq peptide</input>
    </td><td></td></tr>
  <tr><td></td><td>
      <input type="RADIO" name="idtype" value="entrez">  Entrez Gene</input>
    </td><td></td></tr>
  <tr><td></td><td>
      <input type="RADIO" name="idtype" value="hugo">  HUGO Gene Name</input>
    </td><td></td></tr>
  <tr><td></td><td>
      <input type="RADIO" name="idtype" value="cnio">  CNIO ID</input>
    </td><td></td></tr>
  <tr><td></td><td>
      <input type="RADIO" name="idtype" checked value="None">  None of the above</input>
    </td><td></td></tr>
</table>




<br><br>
<input value="Submit" type = "SUBMIT">
<br>
<p style="font-family: 'Bitstream Vera Sans',Verdana,sans">
Push the "Submit" to start execution.</p>
<hr>


<a href="./help/genesrf-help.html" target="help_window">Help</a>




</form>

<hr />


<h3>Source code availability</h3>
<p>See <a href="http://launchpad.net/genesrf/">Launchpad repository for GeneSrF</a> and 
<a href="http://launchpad.net/varselrf/">Launchpad repository for varSelRF</a>. See also the
<a href="http://bioinformatics.org/asterias/wiki/Main/DownloadPage">Bioinformatics.org project page</a>.
</p>

<hr />
<p>
<a href="http://www.python.org/">
<img src="PythonPowered.gif" align=top width=110
     height=44 alt="[Python Powered]" border=0>
</a>

<a href="http://www.r-project.org/">
<img src="Rlogo-5.png" align=top width=100
     height=76 alt="[R Project for Statistical Computing]" border=0>
</a>
</p>
<TABLE width="100%"><TR>
<TD align="right" width="200"><a href="http://asterias.bioinfo.cnio.es"><img src="asterias_app.png" width="200" height="40" border="0"></a></TD>
</TR>
</TABLE>
<address>Send comments to <a href="mailto:rdiaz@cnio.es">webmaster</a></address>
</body> </html>


