<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
<META http-equiv="Content-Type" content="text/html; charset=ISO-8859-1">
<title>Help - GeneSrF</title>
<style type="text/css">
                 body { margin-left: 10%; margin-right: 10%; color: black;} 
<!--                  body { margin-left: 10%; margin-right: 10%; color: black; background="fondo3.jpg";} -->
                 <!-- h1 { margin-left: -8%;} -->
                 h2,h3,h4,h5,h6 { margin-left: -4%; }
div.box { border: solid; border-width: thin; width: 100%; padding: 0.2em; }
div.color {background: rgb(204,204,255);
           padding: 0.5em;
           border: none;
           }
body { font-family: Verdana, sans-serif; }
h1,h2 { font-family: Verdana, sans-serif; }
pre { font-family: monospace; }
               </style>
  </head>
<body background="fondo3.jpg"> 



<br />

<center>
<!-- Creative Commons License -->
<a rel="license" href="http://creativecommons.org/licenses/by-nc-sa/2.0/"><img alt="Creative Commons License" border="0" src="somerights20.gif" /></a><br />
<span class="license">This document is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by-nc-sa/2.0/">Creative Commons License</a></span>.
<!-- /Creative Commons License -->
</center>
<!--

<rdf:RDF xmlns="http://web.resource.org/cc/"
xmlns:dc="http://purl.org/dc/elements/1.1/"
xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"> <Work rdf:about="">
<dc:title>GeneSrF help</dc:title>
<dc:date>1996, 2004</dc:date>
<dc:description>Help for GeneSrF: gene selection using random forests.</dc:description>
<dc:creator><Agent>
<dc:title>Ramón Díaz-Uriarte</dc:title>
</Agent></dc:creator>
<dc:rights><Agent>
<dc:title>Ramón Díaz-Uriarte</dc:title>
</Agent></dc:rights>
<dc:type
rdf:resource="http://purl.org/dc/dcmitype/Text"
/>
<license
rdf:resource="http://creativecommons.org/licenses/by-nc-sa/2.0/"
/>
</Work>

<License
rdf:about="http://creativecommons.org/licenses/by-nc-sa/2.0/">
<permits
rdf:resource="http://web.resource.org/cc/Reproduction"
/>
<permits
rdf:resource="http://web.resource.org/cc/Distribution"
/>
<requires
rdf:resource="http://web.resource.org/cc/Notice"
/>
<requires
rdf:resource="http://web.resource.org/cc/Attribution"
/>
<prohibits
rdf:resource="http://web.resource.org/cc/CommercialUse"
/>
<permits
rdf:resource="http://web.resource.org/cc/DerivativeWorks"
/>
<requires
rdf:resource="http://web.resource.org/cc/ShareAlike"
/>
</License>

</rdf:RDF>

-->
<br />

<hr>
<center>
<h1>GeneSrF</h1>
</center>



<hr>
<ol>
  <li><a href="#intro">Introduction and purpose</a>
  <ul>
    <li><a href="#small">Approach 1: Finding small sets of genes</a></li>
    <li><a href="#large">Approach 2: Finding potentially large sets of genes</a></li>
    <li><a href="#errorstab">Estimates of error rate and stability</a>
    <ul>
      <li><a href="#error.rate">Assessment of error rate</a></li>
      <li><a href="#stability">Stability assessments.</a></li>
    </ul>
  </li>
</ul>
</li>
<li><a href="#usage">Usage</a>
<ul>
  <li><a href="#input">Input</a></li>
  <li><a href="#output">Output</a></li>
  <li><a href="#speed">How long does it take?</a></li>
<li><a href="#examples">Examples</a></li>
</ul>
</li>
<li><a href="#authors">Author and Acknowledgements</a></li>
<li><a href="#terms.use">Terms of use</a></li>
<li><a href="#privacy">Privacy and	Security</a></li>
<li><a href="#warranty">Disclaimer</a></li>  
<li><a href="#refs">References</a></li>
</ol>
<hr>

<h2><a name="intro">Introduction and purpose</a></h2>

<p>GeneSrF is web-based tool for gene selection in classification problems that
uses random forests. Two approaches for gene
    selection are used: one is targeted towards identifying small,
    non-redundant sets of genes that have good predictive performance. The
    second is a more heuristic graphical approach that can be used to identify
    large sets of genes (including redundant genes) related to the outcome of
    interest. The first approach is described in detail in this
<a
href="http://ligarto.org/rdiaz/Papers/rfVS/randomForestVarSel.html">technical
report</a>. The R code is available as an R package from
<a href="http://cran.r-project.org/src/contrib/Descriptions/varSelRF.html">CRAN</a> or
from this <a
href="http://ligarto.org/rdiaz/Software/varSelRF_0.5-4.tar.gz">link</a>.</p>


    <p>Briefly, random forest is an algorithm for classification developed by
    Leo Breiman <cite>Breiman, 2001, 2003</cite> that uses an ensemble of
    classification trees. Each of the classification trees is built using a
    bootstrap sample of the data, and at each split the candidate set of
    variables is a random subset of the variables.  Thus, random forest uses
    both bagging (bootstrap aggregation) and random variable
    selection for tree building.  Each tree is unpruned (grown fully), so as to
    obtain low-bias trees; at the same time, bagging and random variable
    selection result in low correlation of the individual trees.  The algorithm
    yields an ensemble that can achieve both low bias and low variance (from
    averaging over a large ensemble of low-bias, high-variance but low
    correlation trees). Random forests return a prediction as the
	unweighted majority of predictions from a this ensemble of
	classification trees.</p>

    <p>Random forest has several characteristics that make it ideal for gene
    expression data:
    <ul> <li> Can be used when there are many more variables than
    observations.</li>
    <li> Can be used both for two-class and multi-class
    problems of more than two
  classes.</li> <li> Has good predictive performance even when most predictive
  variables are noise, and therefore it does not require a pre-selection of
  genes.</li> <li> Does not overfit.</li> <li> Can handle a mixture of
  categorical and continuous predictors.</li> <li> Incorporates interactions
  among predictor variables.</li> <li> The output is invariant to monotone
  transformations of the predictors.</li> <li> There are high quality and free
  implementations: the original Fortran code from L. Breiman and A. Cutler, and
  an R package from A. Liaw and M. Wiener.</li> <li> Returns measures of
  variable (gene) importance.</li> <li> There is little need to fine-tune
  parameters to achieve excellent performance. </li> </ul>


  <h3><a name="small">Approach 1: Finding small sets of genes</a></h3>
    <p>This approach is described in more detail in the <a
href="http://ligarto.org/rdiaz/Papers/rfVS/randomForestVarSel.html">technical
report</a>. Essentially, we progressively eliminate the least important (where
    importance is based on the measures of variable importance returned by
    random forest itslef) genes until we can obtain no further improvements in
    the "out-of-bag" error rate.  We show in the above report that this
    approach leads to very small sets of genes but good predictive
    accuracy.</p>

    <p>We use the set of parameters that were found to work well in
  <cite><a
  href="http://ligarto.org/rdiaz/Papers/rfVS/randomForestVarSel.html">Díaz-Uriarte
  and Alvarez de Andres (2005)</a></cite>: 2000 trees,
    mtryFactor = 1, se = 1.</p>


  <p>There are other approaches with similar objectives, such as those
  implemented in our tool <a href="http://tnasas.bioinfo.cnio.es">Tnasas</a>:
  with Tnasas we try to
  minimize prediction error but, in most cases, the criteria used for ranking
  genes are univariate criteria.   In addition, the methods implemented in
<a href="http://tnasas.bioinfo.cnio.es">Tnasas</a> are not targeted towards
  identifying potentially large sets of redundant and relevant genes (see next).</p>


<!--   and often the methods implemented in Tnasas  tend to return larger sets  -->

    <h3><a name="large">Approach 2: Finding potentially large sets of genes</a></h3>
    <p>The main objective here is to identify relevant genes for subsequent research; this
  involves obtaining a (probably large) set of genes that are related
  to the outcome of interest, and this set should include genes even if they
  perform similar functions and are highly correlated. Of course, there are
    many other alternatives; for instance, many gene-by-gene approaches, such
    as those implemented in <a
    href="http://pomelo.bioinfo.cnio.es">Pomelo</a>. However, the emphasis of
    the current tool is gene selection in the context of classification
    problems, and the use of variable importance from random forest allows to
    consider interactions and additive relationships between genes (contrary
    to gene-by-gene testing such as in a, say, t-test). </p>

<p>Our main approach here is to plot ordered variable importances, yielding
plots that resemble the ``scree plots'' or ``scree graphs'' common in principal
component analysis. This idea is similar to the the ``importance spectrum''
plots in <cite>Friedman and Meulman</cite>. We want to compare the variable
importance plot from the original data with variable importance plots that are
generated when the class labels and the predictors are independent. Therefore
we permute only the class labels, leaving intact the correlation structure of
the predictors (and of course using the same parameters for random forest). In
this application, the number of random permutations is 50.</p>

<p>Unpublished research with simulated data indicate that these types of plots
can be used to identify even large and highly correlated sets of genes (i.e.,
that the procedure can recover "important genes" even in situations of high
collinearity).</p>



<h3><a name="errorstab">Estimates of error rate and stability</a></h3>
<p>In addition to returning lists of relevant genes, our tool also returns:</p>
<ul>
  <li><a name="error.rate"><b>Assessment of error rate</b></a> of the gene selection method targeted
  towards selecting very small sets of genes. We use the .632+ bootstrap method
  <cite>Efron and Tibshirani, 1997</cite> of the complete gene selection
  procedure, and thus we avoid "selection bias" and related problems
  (<cite>Ambroise and McLachlan, 2002; Simon et al., 2003</cite>). We do not
  perform such a process with the second approach (targeted towards identifying
  potentially large sets of redundant genes) because this later approach is not
  really trying to build a prediction rule, but rather trying to uncover
  "itneresting genes"; moreover, exactly which and how many genes to select is
  something left to the judgment of the user (based on the plots and numbers of
  genes one wants to select). The number of bootstrap replicates we use is 200. </li>

  <li><a name="stability"><b>Stability assessments.</b></a> We provide two such assessemtns.
  <ul>
    <li>How often the selected genes (from Approach 1) are also selected in
    bootstrap runs (with Approach 1).</li>
    <li><a name="sel.prob.plot">Selection probability plots</a>, where we plot, for each of the top ranked
    genes from the original sample, the probability that it is included among the top ranked
    <i>k</i> genes from the (200) bootstrap samples. These plots can be a measure of our
    confidence in choosing the <i>g</i> gene among the top <i>k</i> genes.</li>
  </ul>
  </li>
</ul>


  

    <h2><a name="usage">Usage</a></h2>
    
    <h3><a name="input">Input</a></h3>

    <h4><a name="covariates">Expression data file</a></h4>
    <p>The file with the expression data (the covariates); generally the gene 
    expression data. In this file, rows represent variables (generally genes),
    and columns represent subjects, or samples, or arrays.</p>

    <p> The file for the covariates should be <a name="requirements">formated</a> as:</p> 
    <ul> 
      <li>Data should conform to the "genes in rows,  patients (or arrays) in columns". 
          In other words, each row of the data file is supposed to represent a different gene 
          or variable.</li> 
      <li>Use tab (\t) as the field separator within rows.</li> 
      <li>Use newline or carriage return (\n) between rows. It is also convenient to finish a 
              file with one carriage return (\n).</li> 
      <li>Array names: if you want to name your arrays (useful for the output of
	the analyses) do as follows:
	<ol>
	  <li>Place a line that starts with "&#035;";</li>
	  <li>After the "&#035;" put "Name" or "NAME" or "name" (don't say we
	    don't give you choices);</li>
	  <li>Write the array names (separated by tabs).</li>
	</ol>
      <li>The first column is assumed to contain the ID information for genes, marker, 
        or whatever. This will be used to label the output (but it also means that whatever is 
        in the first column is not used in the analyses).</li> 
      <li>You can have an arbitrary number of rows with comments. These rows must always start 
        with an "&#035;".</li>
	    <li><b>Gene names and array names MUST be unique</b>. If
	      they are not, the program will let you know. If you do
	      not want to provide array names, that is OK, and we will
	      name them with sequential integers starting at 1. If you
	      do not want to provide gene names, then put some
	      arbitrary labels on the first column (e.g., fill it with
	      a sequence of integers).
	      Why do we need gene and array names to be unique?
	      Because in many steps, we need to provide either where
	      we classify a given array (and what should we do if two
	      or more arrays are named "A"?), or the genes used in the
	      classifier (and what should we do if two or more genes
	      are named "gene B"?).</li>

      <li><b>Missing values are NOT allowed</b>. You can use the 
	      <a href="http://gepas.bioinfo.cnio.es/cgi-bin/preprocess">preprocessor</a> and do
	      several things with your data before sending it
	      to Tnasas. We would probably recommend you do imputation after
	      eliminating genes with too many (more than, say, 20&#037;?) missing. Anyway,
	      how best to deal with missing values is not a trivial issue and is outside
	      the scope of this help file.
      
      <li> This is a small covariate data file: 
        <div class="color"> 
          <pre> 
#Name     ge1      ge2      ge1      ge1      ge2 
gene1   23.4    45.6    44      76      85.6 
genW@   3      34      23      56      13 
geneX#  23      25.6    29.4    13.2    1.98   </pre> </div> 
        </li> 

    </ul>
    
    
    <h4><a name="class">Class file</a></h4>
    <p>These are the class labels 
      (e.g., healthy or affected, or different types of cancer) that group the
      samples. Our predictor will try to predict these class labels.</p>
    
	<p>Please note that we do not allow any data set with 3 or fewer cases in any class.
	  Why? Because, on the one hand, any results from such data would be hard to believe;
	  on the other hand, that would result in some cross-validation samples
    having some training samples with 0 elements from one of the classes.</p>
	  
    <p>Separate values by tab (\t), and finish the file with a carriage return or newline.  
      No missing values are allowed here. Class labels can be anything you wish; they 
      can be integers, they can be words, whatever. </p>  
    <p> This is a simple example of class labels file  </p>
    <div class="color"> 
      <pre>  
CL1     CL2     CL1     CL4     CL2     CL2     CL1     CL4       </pre>        </div>       
  


<h4><a name="idc">Type of gene identifier and species</a></h4>
<p>If you use any of the currently standard identifiers for your gene IDs
for either human, mouse, or rat genomes, you can obtain additional
information by clicking on the gene names in the output. This
information is returned from <a
href="http://idclight.bioinfo.cnio.es">IDClight</a> based on that provided
by our <a href="http://idconverter.bioinfo.cnio.es">IDConverter</a>
tool.</p>


    <h3><a name="output">Output</a></h3>
    <h4><a name="outputPlots">Plots</a></h4>
    <h5><a name="f1">OOB error vs. num of genes</a></h5>
    <p>The Out of bag error rate vs. number of genes (number of variables)
    used by random forest. With a thich red line the line for the original
    data, and with dotted black lines the lines from the bootstrap
    samples. Please note that <b>this OOB errors are biased down</b> and should
    not be used as honest estimates of the error rate.</p>
    
    <h5><a name="f2">OOB predictions</a></h5>
    
    <p>For each sample, its (average) out-of-bag class prediction. Thus, this
    is the Out-of-bag prediction for each sample when that sample was not used
    at all for the training process. We provide one such plot for each of the
    class. Thus, for each sample we plot the (average) posterior probability of
    that sample belonging to each class. Obviously, when there are only two
    classes, one plot is like the 1 minus the other plot. You would have
    excellent results if each sample that belongs to a class has an out-of-bag
    (average) posterior probability very close to 1 for its true class and very
    close to zero for all other classes.  </p>

<h5><a name="f3">Importance spectrum plots</a></h5>

<p>Plots of the variable importance of the genes from the original data
compared to variable importances from data sets with the same gene expression
data but randomly permuted class labels. For greater detail, we show the plot
both for all genes and for only the first 200 and 30 genes.</p>

<h5><a name="f4">Selection probability plot</a></h5>

<p>We plot, for each of the top ranked genes from the original sample, the
probability that it is included among the top ranked
    <i>k</i> genes (where in these figures <i>k = 20, 100</i>) from the
    bootstrap samples. These
    plots can be a measure of our
    confidence in choosing the <i>g</i> gene among the top <i>k</i> genes


<h4><a name="resultstext">Results: text</a></h4>

<h5>Variable selection using all data</h5>
<p>The genes selected (variables used) we running the
<a href="#small">gene selection procedure for finding small sets of genes</a>
on the complete, original data.</p>

<h5>Bootstrap results</h5>
<p>The results from the bootstrap run. In particular:</p>
<ul>
  <li><b>Bootstrap estimate of prediction error.</b> This is an honest estimate of the
  error rate.</li>
  <li><b>Error rate at random.</b> This is the (smallest) error you can achieve by betting always on the most common class. In other words, this is the smallest error from using no information from the gene expression.</li>
  <li><b>Number of vars in bootstrapped forests.</b> The number of genes
  selected in the bootstrap runs. We show a summary over the 200 runs,
  including mean, median, 1st and 3rd quantile, maximum and minimum.</li>
  <li><b>Variable freqs. in bootstrapped models.</b> For each of the genes
  (variables) selected in at least one bootstrap run, the frequency with which
  it was selected.</li>
  <li><b>Variable freqs. of variables in forest from all data, and
  summary.</b>For all the genes selected from the run on the complete, original
  data, the frequecy with which those genes appear in the bootstrap runs, as
  well as some summary statistics.</li>
  <li><b>Solutions frequencies in bootstrapped models.</b>"Solution" is how we
  call the actual set of genes selected in a run. This might allow to spot
  certain sets of genes that tend to get selected together, or sets of genes
  that are selected as alternatives to each other.</li>
  <li><b>Mean class membership probabilities from out of bag samples.</b> These
  are the average out-of-bag posterior probabilities ploted <a
  href="#f2">above</a>.
  <li><b>Variable (gene) importances from original data.</b>Gene importances
  for all genes from the original data, before any gene selection. Ordered from
  most to least important.</li>
</ul>

<h4>Download figures and results</h4>
<p>You can download a single compressed file with all figures, in both the
png format showed in the web results as well as in pdf format ---which gives
better quality for printing---,  and
results (a single text file). The format is tar.gz, understood by GNU's tar, standard
in GNU/Linux distributions, and also understood by the common
uncompressors/unzipers/etc in other operating systems.</p>

<h3><a name="speed">How long does it take?</a></h3>
<p>For a data set such as the leukemia data set of Golub et al (1999,
Science, 286: 531-537), with 38
subjects and 3051 genes it takes less than
10 minutes when the servers are lightly loaded, and it takes about 1 hour and 15 minutes
with the Prostate data set of Singh et al. (2002, Cancer Cell, 1: 203--209),
with 102 subjects and 6033 genes. Of course, your mileage will vary.</p>

<h3><a name="examples">Examples</a></h3>
<p>Examples of several runs, one with fully commented results, are available 
<a href="http://genesrf.bioinfo.cnio.es/Examples/index.html">here</a>.


    <h2><a name="authors">Author and Acknowledgements</a></h2>
    
    <p>This program was developped by
      <a href="http://bioinfo.cnio.es/~rdiaz" target="_blank">Ramón Díaz-Uriarte</a>, 
      from the <a href="http://bioinfo.cnio.es" target="_blank">Bioinformatics Unit</a> at
      <a href="http://www.cnio.es" target="_blank">CNIO</a>.
      This tool uses <a href="http://www.python.org">Python</a> for the CGI and
    <a href="http://www.r-project.org">R</a> for the computations.
    The basic squeleton of the R code is the package <a
    href="http://cran.r-project.org/src/contrib/Descriptions/randomForest.html">randomForest</a>
    an R port by Andy Liaw and Matthew Wiener of the <a
    href="http://stat-www.berkeley.edu/users/breiman/RandomForests">Fortran
    original</a> by Leo Breiman and Adele Cutler. The functions for gene
    selection are part of the <a
    href="http://cran.r-project.org/src/contrib/Descriptions/varSelRF.html">varSelRF</a>
    package. The R code also uses the following R packages:
    <a href="http://www.warwick.ac.uk/go/cgiwithr">CGIwithR</a> by
    David Firth,
    <a
    href="http://cran.r-project.org/src/contrib/Descriptions/snow.html">snow</a>,
    by Luke Tierney, A. J. Rossini, Na Li and H. Sevcikova,
    <a href="http://www.stats.uwo.ca/faculty/yu/Rmpi">Rmpi</a>, by Hao Yu,
    and
    <a
    href="http://cran.r-project.org/src/contrib/Descriptions/rsprng.html">rsprng</a>,
    by Na (Michael) Li. </p>
    
    
<p>This application is running on a cluster of machines using <a
href="http://www.debian.org">Debian GNU/Linux</a> as operating system, <a
href="http://www.apache.org">Apache</a> as web server, <a
href="http://www.linuxvirtualserver.org">Linux Virtual Server</a> for web server load-balancing, and <a
href="http://www.lam-mpi.org">LAM/MPI</a> for parallelization.</p>


<p>We want to thank the authors and contributors of these great (and open
source) tools that they have made available for all to use. If you find this
useful, and since R and Bioconductor are developed by a team of volunteers, we
suggest you consider making a donation to the <a
href="http://www.R-project.org/foundation/main.html"> R foundation for
statistical computing</a>.</p>


<p><b>Funding</b> partially provided by Project TIC2003-09331-C02-02 of the
Spanish Ministry of Education and Science. This application is running on a
cluster of machines purchased with funds from the 
<a href="http://www.rticcc.org/">RTICCC</a>.</p> 




     
      
    <h2><a name="terms.use">Terms of use</a></h2>
    <ul>
      <li>You acknowledge that this Software is experimental in nature
	and is supplied "AS IS", without obligation by the authors, the CNIO's
	Bioinformatics Unit or the CNIO to provide accompanying
	services or support. The entire risk as to the quality and performance of the
	Software is with you. The CNIO and the authors expressly disclaim any and all
	warranties regarding the software, whether express or implied, including but
	not limited to warranties pertaining to merchantability or fitness for a
	particular purpose.</li>
      <li>If you use GeneSrF for any publication, we would appreciate if you
	could let us know and if you cite our program (you know, "credit where
	credit is due"). For now, you can give the main web site:
	<a href="http://genesrf.bioinfo.cnio.es">http://genesrf.bioinfo.cnio.es</a>)
	and/or the <a
  href="http://ligarto.org/rdiaz/Papers/rfVS/randomForestVarSel.html">technical report</a>
in that publication.</li>
    <li>We appreciate if you give us feedback concerning bugs, errors or misconfigurations.
	Complaints or suggestions are welcome.</li>
    </ul>
    <br>
    
    <h2><a name="privacy">Privacy and	Security</a></h2>
    <p>Uploaded data set are saved in temporary directories in the server and are
      accessible through the web until they are erased after some time. Anybody can
      access those directories, nevertheless the name of the directories are not
      trivial, thus it is not easy for a third person to access your data.</p>
    <p>In any case, you should keep in mind that communications between the client
      (your computer) and the server are not encripted at all, thus it is also
      possible for somebody else to look at your data while you are uploading or
      dowloading them.</p>
      <br>
      
    <h2><a name="warranty">Disclaimer</a></h2>
    <p>This software is experimental in nature and is supplied "AS IS", without
      obligation by the authors or the CNIO the to provide accompanying services or
      support. The entire risk as to the quality and performance of the software is
      with you. The authors expressly disclaim any and all warranties regarding the
      software, whether express or implied, including but not limited to warranties
      pertaining to merchantability or fitness for a particular purpose.<br>
      
      <!--     (If you are lost using these types of applications, or new, you -->
      <!--       might want to read  -->
      <!--       <a href="http://www.catb.org/~esr/faqs/smart-questions.html"> -->
      <!-- 	How To Ask Questions The Smart Way</a>, by Eric Raymond.) -->
      
      
      
    <h2><a name="refs">References</a></h2>
    
<p>Ambroise C, McLachlan GJ (2002) Selection bias in gene extraction on the basis
    of microarray gene-expression data. 
      <a href="http://www.pnas.org/cgi/content/abstract/99/10/6562">Proc Natl
	Acad Sci USA 99: 6562--6566</a>.</p>

<p>Breiman L (2001) Random forests. <i>Machine Learning</i> 45: 5--32
(<a href="http://oz.Berkeley.edu/users/breiman/randomforest2001.pdf">Tech. report</a>).</p>

<p>Breiman L (2003) <a
      href="http://oz.berkeley.edu/users/breiman/Using_random_forests_v4.0.pdf">
      Manual--Setting Up, Using, And Understanding Random Forests V4.0</a>.</p>


<p>Díaz-Uriarte R, Alvarez de Andrés, S (2005) Gene selection and
classification of microarray data using random forest. In review.
(<a
href="http://ligarto.org/rdiaz/Papers/rfVS/randomForestVarSel.html">tech.
report</a>.)</p>


<p>Efron B, Tibshirani RJ (1997). Improvements on cross-validation: the .632+ bootstrap method.
    <i>J. American Statistical Association</i>, 92: 548-560. </p>


<p>Friedman, JH, Meulman, JJ. (2004). Clustering objects on subsets of
attributes (with discussion). <i>J. Royal Statistical Society, Sr. B</i>, 66: 815--850.</p>

<p>Liaw A, Wiener M (2002) 
    <a href="http://cran.r-project.org/doc/Rnews/Rnews_2002-3.pdf">
Classification and regression by randomForest</a>. <i>R
      News</i>, 2: 18--22.</p>


<p>Simon R, Radmacher MD, Dobbin K, McShane LM (2003) Pitfalls in the use of DNA
  microarray data for diagnostic and prognostic classification. Journal of the
  National Cancer Institute 95: 14--18.</p>



<h3>Copyright</h3>
This document is copyrighted. Copyright © 2005 Ramón Díaz-Uriarte.
<hr>

<!-- Created: Wed Jul 24 15:37:41 CEST 2002 -->
<!-- hhmts start -->Last modified: Thu Aug  4 19:00:07 CEST 2005 <!-- hhmts end -->

<hr>
<p>
      <a href="http://validator.w3.org/check/referer"><img border="0"
          src="valid-html401.png"
          alt="Valid HTML 4.01!" height="31" width="88"></a>
<a href="http://www.anybrowser.org/campaign/">
	<img src="anybrowser3.gif"
	  width="88"
	  height="31"
	  alt="Viewable With Any Browser" border="0"></a>

  </body>
</html>
